### PR TITLE
fix(action-bar): no longer delegates focus when clicked on non-focusable region

### PR DESCRIPTION
**Pull request by anveshmekala**
**Related Issue:** #6462 

## Summary

This PR will fix focusing `calcite-action` in `calcite-action-bar` when user clicks on non-focusable area.  Neither the collapse button nor slotted `calcite-action` will be focused when clicked on non-focusable area.

### CODE REVIEW

File: packages/calcite-components/src/components/action-bar/action-bar.scss [hunk 1]
```diff
@@ -53,5 +56,5 @@
 }
 
 .action-group--bottom {
-  @apply flex-grow justify-end;
```
Comment (driskull) [inline packages/calcite-components/src/components/action-bar/action-bar.scss:56]: Why was this removed?

Comment (anveshmekala) [reply packages/calcite-components/src/components/action-bar/action-bar.scss:56]: With flex-grow, the `calcite-action-group` is filling up the space in full width mode. Any click on the space filled is `delegatingFocu`s in `calcite-action-group`. To avoid this, the empty space will be filled by `calcite-action-bar`  and `delegateFocus` will be removed from `calcite-action-bar`.

**Review by driskull — APPROVED** [review]
Looks great 👍
